app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java

```diff
@@ -70,20 +70,49 @@ protected void onCreate(Bundle savedInstanceState) {
         saveButton = findViewById(R.id.memory_save_button);
         locationInput = findViewById(R.id.memory_change_location_input);
 
-
-        // create a memory with calendar to today
         final Calendar calendar = Calendar.getInstance();
-        ArrayList<Media> media = new ArrayList<>();
-        memory = new Memory(
-                Database.getInstance().newId(),
-                new LatLng(0, 0),
-                new Date(calendar.getTimeInMillis()),
-                "",
-                "",
-                media
-        );
+        if (getIntent().getStringExtra("ID") != null) {
+            isEditMode = true;
+            Database database = Database.getInstance();
+            String sessionId = getIntent().getStringExtra("ID");
+            memory = database.findMemory(sessionId);
+            titleInput.getEditText().setText(memory.getTitle());
+            descriptionInput.getEditText().setText(memory.getDescription());
+            dateInput.getEditText().setText(memory.getDateText());
```

Comment:
Also do `isTitleValid = true ;` and `isDescriptionValid = true;`, because those are definetly valid.

This way the save button will be enabled right away.